src/app/features/tech-record/components/tyres-search/tyres-search.component.ts

```diff
@@ -0,0 +1,171 @@
+    this.globalErrorService.clearErrors();
+    this.route.params.subscribe(p => (this.params = p));
+    this.technicalRecordService.editableTechRecord$.pipe().subscribe(data => (this.viewableTechRecord = data));
+    this.referenceDataService
+      .getTyreSearchReturn$()
+      .pipe(take(1))
+      .subscribe((data: any) => {
+        this.searchResults = data;
+      });
+    if (!this.viewableTechRecord) {
+      this.router.navigate(['../..'], { relativeTo: this.route });
+    }
+  }
+
+  get roles() {
+    return Roles;
+  }
+  get currentVrm(): string | undefined {
+    return this.vehicleTechRecord?.vrms.find(vrm => vrm.isPrimary === true)?.vrm;
+  }
+
+  handleSearch(term: string, filter: string): void {
+    this.globalErrorService.clearErrors();
+    this.searchResults = [];
+    term = term.trim();
+
+    if (!term) {
+      this.globalErrorService.addError({ error: 'You must provide search criteria', anchorLink: 'term' });
+      return;
+    } else if (!filter) {
+      this.globalErrorService.addError({ error: 'You must select a valid search filter', anchorLink: 'term' });
+      return;
+    }
+
+    if (filter === 'code') {
+      this.referenceDataService.loadReferenceDataByKeySearch(ReferenceDataResourceType.Tyres, term);
+    } else {
+      this.referenceDataService.loadTyreReferenceDataByKeySearch(filter, term);
+    }
+
+    this.actions$
+      .pipe(
+        ofType(fetchReferenceDataByKeySearchSuccess, fetchTyreReferenceDataByKeySearchSuccess),
+        mergeMap(_action => this.store.select(selectTyreSearchReturn())),
+        take(1)
+      )
+      .subscribe((data: any) => {
+        this.searchResults = data;
+      });
+  }
+
+  handleSubmit(tyre: ReferenceDataTyre): void {
+    if (this.viewableTechRecord) {
+      const axleIndex = Number(this.params.axleNumber!) - 1;
+      this.viewableTechRecord = cloneDeep(this.viewableTechRecord);
+
+      this.viewableTechRecord!.axles[axleIndex].tyres!.tyreCode = Number(tyre.code);
+      this.viewableTechRecord!.axles[axleIndex].tyres!.tyreSize = tyre.tyreSize;
+      this.viewableTechRecord!.axles[axleIndex].tyres!.plyRating = tyre.plyRating;
+
+      this.store.dispatch(updateEditingTechRecord({ techRecord: this.viewableTechRecord! }));
+      this.router.navigate(['../..'], { relativeTo: this.route });
+    } else {
+      console.error('Unable to update state, changes not saved');
+      /// record relative to previous page lost on refresh - on refresh nav back a page
+    }
+  }
+
+  getErrorByName(errors: GlobalError[], name: string): GlobalError | undefined {
+    return errors.find(error => error.anchorLink === name);
+  }
+
+  cancel() {
+    this.globalErrorService.clearErrors();
+    this.router.navigate(['../..'], { relativeTo: this.route });
+  }
+
+  private pageStart?: number;
+  private pageEnd?: number;
+  public itemsPerPage: number = 10;
+
+  get paginatedFields() {
+    return this.searchResults!.slice(this.pageStart, this.pageEnd);
```

Comment:
`this.searchResults` is initialised to `null` so how can you be sure that it will definitely have an array value? Could the selector not return undefined?

Reply:
Please elaborate. this selector isn't used unless search results returns truthy.